go-sdk/pkg/sdk/keptn.go

```diff
@@ -144,14 +158,15 @@ func (k *Keptn) SendFinishedEvent(event KeptnEvent, result interface{}) error {
 	return k.send(k.createFinishedEventForTriggeredEvent(inputCE, result))
 }
 
-func (k *Keptn) gotEvent(event cloudevents.Event) {
+func (k *Keptn) gotEvent(ctx context.Context, event cloudevents.Event) {
 	if !keptnv2.IsTaskEventType(event.Type()) {
 		log.Errorf("event with event type %s is no valid keptn task event type", event.Type())
 		return
 	}
-
+	ctx.Value("Wg").(*sync.WaitGroup).Add(1)
```

Comment:
the "Wg" string key is used on multiple places...
Suggestion: use an "opaque" type for the key like:

```
// Opaque key type used for graceful shutdown context value
type gracefulShutdownKeyType struct{}
var gracefulShutdownKey = gracefulShutdownKeyType{}
```
you can place this somewhere in `keptn.go`
As key type you just use `ctx.Value(gracefulShutdownKey)...`